lib/fizzy/span.hpp

```diff
@@ -0,0 +1,55 @@
+// Fizzy: A fast WebAssembly interpreter
+// Copyright 2019-2020 The Fizzy Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <iterator>
+#include <type_traits>
+
+namespace fizzy
+{
+/// The span describes an object that can refer to a contiguous sequence of objects with the first
+/// element of the sequence at position zero.
+///
+/// This is minimal implementation of C++20's std::span:
+/// https://en.cppreference.com/w/cpp/container/span
+/// Only `const T` is supported.
+template <typename T, typename = typename std::enable_if_t<std::is_const_v<T>>>
+class span
+{
+    T* const m_begin = nullptr;
+    const std::size_t m_size = 0;
+
+public:
+    using value_type = std::remove_cv_t<T>;
+    using iterator = T*;
+    using reverse_iterator = std::reverse_iterator<iterator>;
+
+    constexpr span() noexcept = default;
+    constexpr span(const span&) = default;
+
+    constexpr span(const T* begin, std::size_t size) noexcept : m_begin{begin}, m_size{size} {}
```

Comment:
if `T` is already `const`, is `const` needed here or how is it allowed at all?
```suggestion
    constexpr span(T* begin, std::size_t size) noexcept : m_begin{begin}, m_size{size} {}
```

Reply:
Here I mean it looks at least inconsistent with the `m_begin` definition above, it should be either both `T*` or both `const T*`.
But not sure if it would somehow break anything?

Reply:
Oh yes, fixed.